content/providers/01-ai-sdk-providers/12-black-forest-labs.mdx

```diff
@@ -0,0 +1,147 @@
+---
+title: Black Forest Labs
+description: Learn how to use Black Forest Labs models with the AI SDK.
+---
+
+# Black Forest Labs Provider
+
+[Black Forest Labs](https://bfl.ai/) provides a generative image platform for developers with FLUX-based models. Their platform offers fast, high quality, and in-context image generation and editing with precise and coherent results.
+
+## Setup
+
+The Black Forest Labs provider is available via the `@ai-sdk/black-forest-labs` module. You can install it with
+
+<Tabs items={['pnpm', 'npm', 'yarn', 'bun']}>
+  <Tab>
+    <Snippet text="pnpm add @ai-sdk/black-forest-labs" dark />
+  </Tab>
+  <Tab>
+    <Snippet text="npm install @ai-sdk/black-forest-labs" dark />
+  </Tab>
+  <Tab>
+    <Snippet text="yarn add @ai-sdk/black-forest-labs" dark />
+  </Tab>
+
+  <Tab>
+    <Snippet text="bun add @ai-sdk/black-forest-labs" dark />
+  </Tab>
+</Tabs>
+
+## Provider Instance
+
+You can import the default provider instance `blackForestLabs` from `@ai-sdk/black-forest-labs`:
+
+```ts
+import { blackForestLabs } from '@ai-sdk/black-forest-labs';
+```
+
+If you need a customized setup, you can import `createBlackForestLabs` and create a provider instance with your settings:
+
+```ts
+import { createBlackForestLabs } from '@ai-sdk/black-forest-labs';
+
+const blackForestLabs = createBlackForestLabs({
+  apiKey: 'your-api-key', // optional, defaults to BFL_API_KEY environment variable
+  baseURL: 'custom-url', // optional
+  headers: {
+    /* custom headers */
+  }, // optional
+});
+```
+
+You can use the following optional settings to customize the Black Forest Labs provider instance:
+
+- **baseURL** _string_
+
+  Use a different URL prefix for API calls, e.g. to use a regional endpoint.
+  The default prefix is `https://api.bfl.ai/v1`.
+
+- **apiKey** _string_
+
+  API key that is being sent using the `x-key` header.
+  It defaults to the `BFL_API_KEY` environment variable.
+
+- **headers** _Record&lt;string,string&gt;_
+
+  Custom headers to include in the requests.
+
+- **fetch** _(input: RequestInfo, init?: RequestInit) => Promise&lt;Response&gt;_
+
+  Custom [fetch](https://developer.mozilla.org/en-US/docs/Web/API/fetch) implementation.
+  You can use it as a middleware to intercept requests,
+  or to provide a custom fetch implementation for e.g. testing.
+
+## Image Models
+
+You can create Black Forest Labs image models using the `.image()` factory method.
+For more on image generation with the AI SDK see [generateImage()](/docs/reference/ai-sdk-core/generate-image).
+
+### Basic Usage
+
+```ts
+import { blackForestLabs } from '@ai-sdk/black-forest-labs';
+import { experimental_generateImage as generateImage } from 'ai';
+import fs from 'fs';
```

Comment:
```suggestion
import { writeFileSync } from "node:fs";
import { blackForestLabs } from "@ai-sdk/black-forest-labs";
import { experimental_generateImage as generateImage } from "ai";
```

Reply:
this is probably copied from another docs page, I can fix the other docs pages in a separate PR